### PR TITLE
add a funnel visualization type

### LINKED ISSUE #9098: funnels UI

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

* Adds initial support for event funnel charts.
Funnel applies to events, grouped by `secure_session_id`, to allow charting sessions where a number of events happen.
Once events have additional session metadata, it should be possible to group by user rather than by session.
Currently, some of the graph configuration is disabled when the chart is a funnel, as it is set to the
default values that the funnel querying assumes (ie. the group by / bucket by / metric)

The funnel loads by using the `Metrics` query, sharing the query logic with other graph types.
Multiple steps are loaded serially to allow filtering on the group values (sessions) from the previous bar.

Closes #9098
Closes HIG-4864

## How did you test this change?

Screenshot 2024-10-07 at 10 21 05

- [x] Switching to a funnel chart when the selected source is `events`
- [x] Adding / removing steps using event filters, setting different titles for various steps
- [x] Saving / reloading dashboard with event filters 
- [x] Loading state for the funnel is correct, showing funnel once all steps are loaded
- [x] UI is using correct figma designs

## Are there any deployment considerations?

no

## Does this work require review from our design team?

will review with @julian-highlight

### CODE REVIEW

**Comment by Vadman97** [comment]
should be ready for review @SpennyNDaJets

File: frontend/src/pages/Graphing/components/FunnelChart.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+					</Box>
+				))}
+			</Box>
+		)
+	}
+
+export type FunnelChartConfig = {
+	type: 'Funnel chart'
+	showLegend: boolean
+	display?: FunnelDisplay
```
Comment (SpennyNDaJets) [inline frontend/src/pages/Graphing/components/FunnelChart.tsx:78]: This is really cool, noticed the colors are lacking a little in the vertical view, and wanted to make sure the shape was anticipated
Screenshot 2024-10-10 at 11 02 50 AM

**Review by SpennyNDaJets — APPROVED** [review]
Looking good! Nothing too blocking, but maybe the bug related to making trace/log funnels?

File: frontend/src/pages/Graphing/EventSelection/EventSteps.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	return (
+		<>
+			{steps.map((step, index) => (
+				<EventStep
+					key={index}
+					index={index}
+					step={step}
+					startDate={startDate}
+					endDate={endDate}
+					onRemove={() =>
```
[the remark this answers is not in the document]
Comment (Vadman97) [reply frontend/src/pages/Graphing/EventSelection/EventSteps.tsx:178]: yup! discussed this w @julian-highlight - will follow up on this

File: __generated/rr/rrweb/rr.js
[the remark this answers is not in the document]
Comment (Vadman97) [reply __generated/rr/rrweb/rr.js:1]: merge conflict in this case where i had another branch build rrweb (newer submodule) and then reflame-build which bundles the rrweb bundle into a js file

File: frontend/src/pages/Graphing/components/FunnelChart.tsx (hunk 1 above)
Comment (Vadman97) [reply frontend/src/pages/Graphing/components/FunnelChart.tsx:78]: having chatted w. julian, we're going to hide this view for now and instead allow bar vs. line graph for the funnel

**Pull request merged**